client/src/test/java/io/pravega/client/auth/CredentialsHelperTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.auth;
+
+import com.google.common.collect.ImmutableMap;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.Credentials;
+import io.pravega.test.common.AssertExtensions;
+import java.util.Map;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class CredentialsHelperTest {
+    @Test
+    public void testextractCredentials() {
+        //No creds defined
+        ClientConfig config = ClientConfig.builder().build();
+        config = CredentialsHelper.extractCredentials(config);
+        assertEquals("Empty list should return null",
+                config.getCredentials(), null);
+
+        //Test custom creds
+        System.setProperty("pravega.client.auth.method", "temp");
```

Comment:
Unit tests should NOT have side effects. Please do not set System Properties in your tests. 

An alternative is to have your main class define a custom, test-only constructor, that accepts 2 maps, one for sys props and one for env vars.

Reply:
Clearing the side effects at the end of the test cases.

Reply:
The way you are clearing them is incorrect for two reasons
1. If there was a value before that, you will have removed it, thus cause side effects.
2. If your test fails, then the cleanup will not execute.

Please convert your class from an all-static-methods class to an instance class, with a default constructor that picks up from `System.getProperties()` and `System.getEnv()`, and one which takes two maps. That is way cleaner and easier to test.